src/sqlfluff/core/config.py

```diff
@@ -755,9 +755,13 @@ def __init__(
             ("exclude_rules", "rule_denylist"),
         ]:
             if self._configs["core"].get(in_key, None):
-                self._configs["core"][out_key] = _split_comma_separated_string(
-                    self._configs["core"][in_key]
-                )
+                # Checking if key is string as can potentially be a list to
+                if isinstance(self._configs["core"][in_key], str):
+                    self._configs["core"][out_key] = _split_comma_separated_string(
+                        self._configs["core"][in_key]
+                    )
+                else:
+                    self._configs["core"][out_key] = self._configs["core"][in_key]
```

Comment:
Should this also check that the resulting config value is a list and raise an error otherwise?

Also, this makes the config for these values inconsistent with certain rule configs like `ignore_words` for rule L010. I wonder if this logic should get moved into `_split_comma_separated_string` and to make `BaseRule.split_comma_separate_string` call that as well so they share the logic.

Something like this maybe?

```python
def _split_comma_separated_string(raw: Union[str, List[str]]) -> List[str]:
    if isinstance(raw, str):
        return [s.strip() for s in raw.split(",") if s.strip()]
    if isinstance(raw, list):
        return raw
    raise SQLFluffUserError(f"Expected list or comma separated string. Got {type(raw)} instead for value {raw}.")
```